pkg/eks/tasks.go

```diff
@@ -280,6 +282,37 @@ func (c *ClusterProvider) CreateExtraClusterConfigTasks(ctx context.Context, cfg
 		newTasks.Append(identityproviders.NewAssociateProvidersTask(ctx, *cfg.Metadata, cfg.IdentityProviders, c.AWSProvider.EKS()))
 	}
 
+	if len(cfg.IAMIdentityMappings) > 0 {
+		newTasks.Append(&tasks.GenericTask{
+			Description: "create iamIdentitymappings",
+			Doer: func() error {
+				clientSet, err := c.NewStdClientSet(cfg)
+				if err != nil {
+					return errors.Wrap(err, "error creating Clientset")
+				}
+
+				rawClient, err := c.NewRawClient(cfg)
+				if err != nil {
+					return errors.Wrap(err, "error creating rawClient")
+				}
+				m, err := iamidentitymapping.New(cfg, clientSet, rawClient, cfg.Metadata.Region)
+				if err != nil {
+					return errors.Wrap(err, "error initilising iamidentitymapping")
+				}
+
+				for _, mapping := range cfg.IAMIdentityMappings {
+					if err := mapping.Validate(); err != nil {
```

Comment:
This validation should happen sooner, before any tasks are run. Can you move this to `ValidateClusterConfig`?